From: Rasesh Mody <rmody@brocade.com>
Date: Tue, 17 Dec 2013 17:07:31 -0800
Subject: bna: Add software timestamping support
Patch-mainline: v3.14-rc1
Git-commit: fee1253e280a6114688958f4762a838d4d9bc306
References: bnc#855232 FATE#315938

- Invoke skb_tx_timestamp() API just before invoking txq_doorbell()
 - Add ethtool (-T) support

Signed-off-by: Rasesh Mody <rmody@brocade.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/brocade/bna/bnad.c         |    2 ++
 drivers/net/ethernet/brocade/bna/bnad_ethtool.c |    1 +
 2 files changed, 3 insertions(+)
--- a/drivers/net/ethernet/brocade/bna/bnad.c
+++ b/drivers/net/ethernet/brocade/bna/bnad.c
@@ -2912,6 +2912,8 @@ bnad_start_xmit(struct sk_buff *skb, str
 	if (unlikely(!test_bit(BNAD_TXQ_TX_STARTED, &tcb->flags)))
 		return NETDEV_TX_OK;
 
+	skb_tx_timestamp(skb);
+
 	bna_txq_prod_indx_doorbell(tcb);
 
 	return NETDEV_TX_OK;
--- a/drivers/net/ethernet/brocade/bna/bnad_ethtool.c
+++ b/drivers/net/ethernet/brocade/bna/bnad_ethtool.c
@@ -1131,6 +1131,7 @@ static const struct ethtool_ops bnad_eth
 	.get_eeprom = bnad_get_eeprom,
 	.set_eeprom = bnad_set_eeprom,
 	.flash_device = bnad_flash_device,
+	.get_ts_info = ethtool_op_get_ts_info,
 };
 
 void
